File: PracticasPython/Practica7/JuegoAdivinaImpar.py
from JuegoAdivinaNumero import JuegoAdivinaNumero
class JuegoAdivinaImpar(JuegoAdivinaNumero):
    def __init__(self, numeroDeVidas, record, numeroAAdivinar):
        super().__init__(numeroDeVidas, record, numeroAAdivinar)
    def validaNumero(self, n):
            if 1 <= n <= 10 and n%2 == 1:
                return True
            elif 1 <= n <= 10 and n%2 != 1:
                print("ERROR")
                return False
            else:
                return False

File: PracticasJava/Practica7/src/JuegoAdivinaImpar.java


public class JuegoAdivinaImpar extends JuegoAdivinaNumero {
	public JuegoAdivinaImpar(int numeroDeVidas, int record, int numeroAAdivinar) {
        super(numeroDeVidas, record, numeroAAdivinar);
    }
	@Override
	public boolean validaNumero(int n) {
		if (n > (-1) && n < 11 && n % 2== 1) {
			return true;
		} else if (n > (-1) && n < 11 && n % 2== 0) {
			System.out.println("ERROR");
			return false;
		} else {
            return false;  
        }
	}
}

